src/blueprint/from_config.rs

```diff
@@ -358,8 +373,41 @@ fn update_modify(
     None => Valid::Ok(Some(b_field)),
   }
 }
+fn update_const_field(field: &config::Field, mut b_field: FieldDefinition, config: &Config) -> Valid<FieldDefinition> {
+  match field.const_field.as_ref() {
+    Some(const_field) => {
+      let mut data = const_field.data.to_owned();
+      // Add {key: null} pair if key is nullable and not provided in const value
+      if let Some(typ) = config.find_type(&field.type_of) {
+        if let Value::Object(ref mut map) = data {
+          for key in typ.fields.keys() {
+            if let Some(field) = typ.fields.get(key) {
+              let json_has_key = map.contains_key(key);
+              let field_required = field.required;
+              if !json_has_key && !field_required {
+                map.insert(key.to_owned(), Value::Null);
+              }
+            }
+          }
+        }
+      }
+      if let Ok(gql_value) = ConstValue::from_json(data.to_owned()) {
+        match to_json_schema_for_field(field, config).validate(&gql_value) {
+          Ok(_) => {
+            b_field.resolver = Some(Literal(data.to_owned()));
+            Valid::Ok(b_field)
+          }
+          Err(err) => err.into(),
+        }
+      } else {
+        Valid::fail("@const data: invalid graphql value".to_owned())
```

Comment:
Not a very useful error message. It should say invalid JSON or something, and should use the error information available in the `Err`.